src/test/kotlin/org/jetbrains/research/transformations/AnonymizationTest.kt

```diff
@@ -0,0 +1,62 @@
+package org.jetbrains.research.transformations
+
+import com.github.gumtreediff.gen.python.PythonTreeGenerator
+import com.github.gumtreediff.tree.TreeContext
+import org.jetbrains.research.transformations.util.ParserSetup
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.Test
+
+internal class AnonymizationTest {
+
+    @Test
+    fun checkVariableAnonymizationForTest_1() {
+        ParserSetup.checkSetup()
+
+        val srcFile = javaClass.getResource("test_1.py").path
+        val treeCtx : TreeContext = PythonTreeGenerator().generateFromFile(srcFile)
+
+        Anonymization.apply(treeCtx, true)
+        val tree = treeCtx.root
+        val treeIterator = tree.trees.iterator()
+
+
+        while (treeIterator.hasNext()) {
+            val node = treeIterator.next()
+            if (treeCtx.getTypeLabel(node) == "NameStore") {
+                val nodeMap = node.getMetadata(Anonymization.metadataKey) as MutableMap<String, String>
+                if (nodeMap.containsKey("v0")) {
+                    assertEquals("a", nodeMap["v0"])
+                }
+                if (nodeMap.containsKey("v1")) {
+                    assertEquals("b", nodeMap["v1"])
+                }
+                if (nodeMap.containsKey("v2")) {
+                    assertEquals("c", nodeMap["v2"])
+                }
+            }
+        }
```

Comment:
Please, don't use several asserts in one test. There is a great description of how good tests should be written, take a look https://phauer.com/2019/modern-best-practices-testing-java/

Note several useful ideas that we definitely want to see (detailed descriptions can be found in the article above):
- use words **expected** and **actual** in variable names
- keep tests clean and simple, moving as much as possible to helper functions with comprehensible names
- use parametrized tests instead of looping
- use Kotlin’s Backticks to describe what test should do

You can also read https://phauer.com/2018/best-practices-unit-testing-kotlin/ :)


My suggestion for transformation testing is to create pairs of files:
source_1.py - anonymized_source_1.py
and check if the code after applying a transformation to source_1.py is equal to anonymized_source_1.py
To test reverse apply, you can apply a transformation to source_1.py, make reverse apply and that it's equal to source_1.py

Note that there will be more transformations so to not mess it up you can create a separated folder in resources:

- /src/test/resources/org/jetbrains/research/transformations/anonimazation/
-- classes
---- source_1.py
---- source_2.py
---- source_3.py
---- anonimized_source_1.py
---- anonimized_source_2.py
---- anonimized_source_3.py
-- variables
---- .....
-- methods
---- ......
-- arguments 
---- .....

and so on